users/service.go

```diff
@@ -189,7 +189,21 @@ func (svc service) ListClients(ctx context.Context, token string, pm mgclients.P
 		return pg, err
 	}
 
-	pg, err := svc.clients.SearchClients(ctx, pm)
+	// Return empty page if user is not super admin
+	// And search criteria by identity
+	if pm.Identity != "" {
+		return mgclients.ClientsPage{}, nil
+	}
+
```

Comment:
We can remove this lines, since we are creating new page in below line with omitting the identity from actual page request 
```suggestion
```

Reply:
Then we should check for other query params like `tag` and `metadata`
So code becomes like

```go
// Return empty page if user is not super admin
// And search criteria by identity
if pm.Identity != "" {
	return mgclients.ClientsPage{}, nil
}

// Return empty page if user is not super admin
// And search criteria by identity
if pm.Tag != "" { 
    return mgclients.ClientsPage{}, nil
}

// Return empty page if user is not super admin
// And search criteria by identity

if pm.Metadata != "" { 
    return mgclients.ClientsPage{}, nil
}
``` 
This will looks like validation of input. 

So @dborovcanin how we can proceed ? 

At present known options 
- Validation of inputs in service for non-admins.
- Having seperate serach endpoint.

If you have any other idea , please let us know here

Reply:
Let's switch to the search endpoint.